src/components/checkout/CheckoutBlock.tsx

```diff
@@ -166,6 +173,22 @@ const CheckoutBlock: React.VFC<{
     const isValidShipping = !hasPhysicalProduct || validateShipping(shipping)
     const isValidInvoice = validateInvoice(invoice).length === 0
 
+    refetchCartProductsWithInventory()
+    if (
+      !loadingCartProductsWithInventory &&
+      cartProductsWithInventory.some(cartProduct => cartProduct.buyableQuantity === 0)
+    ) {
```

Comment:
should use `await` operator if you want to refetch first

Reply:
as the code review said, plz use `handleCheckoutAsync` instead